src/main/java/org/threadly/litesockets/tcp/SSLClient.java

```diff
@@ -338,8 +316,8 @@ public void writeForce(ByteBuffer buffer) {
               super.writeForce(tmpBB);
             }
           } catch(Exception e) {
-            if(!handshakeFuture.isDone()) {
-              handshakeFuture.setFailure(e);
+            if(! handshakeFuture.isDone() && 
```

Comment:
There is no reason to check done here any more.  That will happen when setting the failure.